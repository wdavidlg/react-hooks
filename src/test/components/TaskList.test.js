const { shallow } = require("enzyme")
const { default: TaskList } = require("../../components/08-useReducer/TaskList")
const { tasks } = require("./fixtures/tasks")


describe('Pruebas con <TaskList />', () => {

    const complete = jest.fn();
    const handleDelete = jest.fn();

    const wrapper = shallow(
        <TaskList 
            tasks={tasks}
            complete={complete}
            handleDelete={handleDelete}
        />
    );
    test('debe de mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener todos los elementos', () => {
        
        const tam = wrapper.find('Task').length;
        expect(tam).toBe(tasks.length);
        expect( typeof wrapper.find('Task').at(0).prop('handleDelete')).toBe( 'function' );
    })

})
