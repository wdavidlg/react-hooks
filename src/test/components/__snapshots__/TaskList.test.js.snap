// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <TaskList /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <ul
    className="list-group list-group-flush"
  >
    <Task
      complete={[MockFunction]}
      handleDelete={[MockFunction]}
      i={0}
      key="1"
      task={
        Object {
          "done": false,
          "id": 1,
          "task": "Aprender React",
        }
      }
    />
    <Task
      complete={[MockFunction]}
      handleDelete={[MockFunction]}
      i={1}
      key="2"
      task={
        Object {
          "done": false,
          "id": 2,
          "task": "Aprender Node",
        }
      }
    />
  </ul>
</Fragment>
`;
